### PR TITLE
feat(hooks): add Stop hook for backlog task quality gate

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

Implements a Stop hook that enforces backlog task quality gates before PR creation. This ensures tasks are marked "Done" with all acceptance criteria complete before the conversation ends, maintaining backlog accuracy.

## Changes

- **Stop Hook Implementation** (`.claude/hooks/stop-quality-gate.py`):
  - Detects PR creation intent in conversation summary using comprehensive regex patterns
  - Checks for "In Progress" tasks via `backlog task list` CLI
  - Blocks session stop if incomplete tasks exist, providing actionable guidance
  - Follows fail-open principle for reliability (allows stop on any error)
  - Supports force/skip bypass for power users
  - 5-second timeout for CLI operations

- **Comprehensive Test Suite** (`.claude/hooks/test-stop-quality-gate.py`):
  - 10+ test cases covering edge cases and error conditions
  - Tests PR detection patterns, task state checking, fail-open behavior
  - Validates JSON protocol compliance

- **Configuration** (`.claude/settings.json`):
  - Registers Stop hook with 10s timeout
  - Integrates with existing hook infrastructure

- **Documentation**:
  - ADR-003 documents design decisions and implementation details
  - Updated CLAUDE.md with Stop hook description
  - Comprehensive edge case handling documented

## Design Decisions (ADR-003)

**Why Stop Hook over Pre-Tool-Use?**
- Catches PR intent before session ends (natural checkpoint)
- Works with all PR creation methods (CLI, web UI, etc.)
- Clear fail-open semantics at session boundary
- User can review guidance before ending session

**Quality Gate Logic**:
1. Parse conversation for PR intent (9 detection patterns)
2. If PR detected → check for "In Progress" tasks
3. If incomplete tasks → block with clear guidance listing tasks
4. On any error → fail-open (allow stop)

**Fail-Open Scenarios**:
- Backlog CLI not available or times out (>5s)
- Invalid JSON input or parsing errors
- Any unexpected exception

## Testing

All tests pass with comprehensive coverage:
- PR detection across various phrases
- Task state checking and blocking logic
- Fail-open error handling
- Force/skip bypass functionality
- Edge cases (empty context, CLI errors, multiple tasks)

## CI Fixes in This PR

This PR addresses the previous CI failures:

1. **Lint**: ✅ All `ruff check` tests pass
2. **DCO Sign-off**: ✅ All commits now have `Signed-off-by` lines
   - Used `git filter-branch` to add DCO sign-offs to all 15 commits
   - Preserved original commit messages and co-author attributions

## Task Completion

Completes backlog task-189:
- ✅ Implement Stop hook with PR detection
- ✅ Integrate backlog CLI with timeout handling
- ✅ Add comprehensive test suite
- ✅ Document design decisions in ADR
- ✅ Update CLAUDE.md

Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: .claude/hooks/test-stop-quality-gate.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+                if "Test task for stop hook" in line:
+                    # Extract task ID from format: [HIGH] task-123 - Title
+                    import re
+
+                    match = re.search(r"task-\d+", line)
+                    if match:
+                        task_id = match.group()
+                        subprocess.run(
+                            ["backlog", "task", "archive", task_id], capture_output=True
+                        )
```
[the remark this answers is not in the document]
Comment (jpoley) [reply .claude/hooks/test-stop-quality-gate.py]: fix me
